envisage/plugins/ipython_kernel/internal_ipkernel.py

```diff
@@ -77,15 +78,37 @@ class InternalIPKernel(HasStrictTraits):
     #: This is a list of tuples (name, value).
     initial_namespace = List()
 
-    def init_ipkernel(self, gui_backend):
+    def init_ipkernel(self, gui_backend=None):
         """ Initialize the IPython kernel.
 
         Parameters
         ----------
-        gui_backend -- string
-          The GUI mode used to initialize the GUI mode. For options, see
-          the `ipython --gui` help pages.
+        gui_backend -- string, optional
+            The GUI mode used to initialize the GUI event loop integration. For
+            options, see the `ipython --gui` help pages. If not given, no event
+            loop integration is set up.
+
+        .. note:: Use of this argument is deprecated!
         """
+        # For backwards compatibility, we allow a kernel to be initialised
+        # twice, and we ignore the second initialization, but warn.
+        if self.ipkernel is not None:
+            warnings.warn(
+                "IPython kernel is being initialised for a second time. "
+                "This may become an error in future versions of this package.",
+                DeprecationWarning,
+            )
+            return
+
+        if gui_backend is not None:
+            warnings.warn(
+                "The gui_backend argument is deprecated. "
+                "Integration with a GUI event loop can be "
+                "achieved by setting the 'eventloop' attribute on the "
+                "kernel instance",
+                DeprecationWarning,
+            )
+
         # Start IPython kernel with GUI event loop support
         self.ipkernel = gui_kernel(gui_backend)
```

Comment:
does it make sense to also update `gui_kernel` to make `gui_backend` an optional argument?

Reply:
I don't think so; as far as I'm aware, no-one is using `gui_kernel` directly (but I'm happy to be corrected on that point).

Reply:
Though possibly we should make `gui_kernel` private in this PR.

Reply:
Done.